instrumentation/influxdb-2.4/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/influxdb/v2_4/InfluxDbClientTest.java

```diff
@@ -0,0 +1,311 @@
+                span ->
+                    span.hasName("SELECT " + databaseName)
+                        .hasKind(SpanKind.CLIENT)
+                        .hasAttributesSatisfying(
+                            attributeAssertions(
+                                "SELECT * FROM cpu_load where test1 = ?",
+                                "SELECT",
+                                databaseName))));
+  }
+
+  @Test
+  void testQueryWithThreeArguments() throws InterruptedException {
+    Query query =
+        new Query(
+            "SELECT * FROM cpu_load where time >= '2022-01-01T08:00:00Z' AND time <= '2022-01-01T20:00:00Z'",
+            databaseName);
+    BlockingQueue<QueryResult> queue = new LinkedBlockingQueue<>();
+
+    influxDb.query(query, 2, result -> queue.add(result));
+    Thread.sleep(2000);
+    queue.poll(20, TimeUnit.SECONDS);
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("SELECT " + databaseName)
+                        .hasKind(SpanKind.CLIENT)
+                        .hasAttributesSatisfying(
+                            attributeAssertions(
+                                "SELECT * FROM cpu_load where time >= ? AND time <= ?",
+                                "SELECT",
+                                databaseName))));
+  }
+
+  @Test
+  void testQueryWithThreeArgumentsCallback() throws InterruptedException {
+    Query query = new Query("SELECT * FROM cpu_load", databaseName);
+    BlockingQueue<QueryResult> queue = new LinkedBlockingQueue<>();
+
+    influxDb.query(query, 2, result -> queue.add(result));
+    Thread.sleep(2000);
+    queue.poll(20, TimeUnit.SECONDS);
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span ->
+                    span.hasName("SELECT " + databaseName)
+                        .hasKind(SpanKind.CLIENT)
+                        .hasAttributesSatisfying(
+                            attributeAssertions(
+                                "SELECT * FROM cpu_load", "SELECT", databaseName))));
+  }
+
+  @Test
+  void testQueryWithFiveArguments() throws InterruptedException {
+    CountDownLatch countDownLatch = new CountDownLatch(1);
+    CountDownLatch countDownLatchFailure = new CountDownLatch(1);
+    Query query =
+        new Query(
+            "SELECT MEAN(water_level) FROM h2o_feet where time = '2022-01-01T08:00:00Z'; SELECT water_level FROM h2o_feet LIMIT 2",
+            databaseName);
+    influxDb.query(
+        query,
+        10,
+        (cancellable, queryResult) -> {
+          countDownLatch.countDown();
+        },
+        () -> {},
+        throwable -> {
+          countDownLatchFailure.countDown();
+        });
+    Assertions.assertFalse(countDownLatchFailure.await(1, TimeUnit.SECONDS));
+    Assertions.assertTrue(countDownLatch.await(1, TimeUnit.SECONDS));
```

Comment:
Short timeout could result in flaky tests. I'd use a 10 second timeout for the count down latch that is supposed to be called. Maybe it would make sense to test the failed callback separately? Would an invalid query trigger it?